core/trino-main/src/main/java/io/trino/operator/FlatHash.java

```diff
@@ -153,13 +153,36 @@ public int getCapacity()
         return capacity;
     }
 
+    /**
+     * Releases memory associated with the hash table which is no longer necessary to produce output. Subsequent
+     * calls to insert new elements are rejected, and calls to {@link FlatHash#appendTo(int, BlockBuilder[])} will
+     * incrementally release memory associated with prior groupId values assuming that the caller will only call into
+     * the method to produce output in a sequential fashion.
+     */
+    public void startReleasingOutput()
+    {
+        if (isReleasingOutput()) {
```

Comment:
A better place for it might be `BigintGroupByHash` and `FlatGroupByHash` since they also release internal structures

Reply:
This is the only place where we have structures released that definitively prove that we've transitioned into releasing output. In `FlatGroupByHash` and `BigintGroupByHash` we null out values that are lazily initialized and therefore are allowed to be `null` without necessarily indicating that a terminal transition to releasing output has occurred.